File: src/test/java/de/pflugradts/passbird/domain/model/ddd/AggregateRootTest.java
package de.pflugradts.passbird.domain.model.ddd;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class AggregateRootTest {

    @Test
    void shouldRegisterDomainEvent() {
        // given
        final var domainEvent = new DomainEvent(){};
        final var aggregate = setupAggregate();

        // when
        aggregate.registerDomainEvent(domainEvent);

        // then
        assertThat(aggregate)
                .extracting(AggregateRoot::getDomainEvents).isNotNull()
                .asList().containsExactly(domainEvent);
    }

    @Test
    void shouldClearDomainEvent() {
        // given
        final var domainEvent = new DomainEvent(){};
        final var aggregate = setupAggregate();

        // when
        aggregate.registerDomainEvent(domainEvent);
        aggregate.clearDomainEvents();

        // then
        assertThat(aggregate)
                .extracting(AggregateRoot::getDomainEvents).isNotNull()
                .asList().isEmpty();
    }

    private AggregateRoot setupAggregate() {
        return new AggregateRoot() {
        };
    }

}

File: src/test/kotlin/de/pflugradts/passbird/domain/model/ddd/AggregateRootTest.kt
package de.pflugradts.passbird.domain.model.ddd

import org.junit.jupiter.api.Test
import strikt.api.expectThat
import strikt.assertions.containsExactly
import strikt.assertions.isEmpty

class AggregateRootTest {
    @Test
    fun `should register domain event`() {
        // given
        val domainEvent = object : DomainEvent {}
        val aggregate = object : AggregateRoot() {}

        // when
        aggregate.registerDomainEvent(domainEvent)

        // then
        expectThat(aggregate.getDomainEvents()).containsExactly(domainEvent)
    }

    @Test
    fun `should clear domain event`() {
        // given
        val domainEvent = object : DomainEvent {}
        val aggregate = object : AggregateRoot() {}

        // when
        aggregate.registerDomainEvent(domainEvent)
        aggregate.clearDomainEvents()

        // then
        expectThat(aggregate.getDomainEvents()).isEmpty()
    }
}
